packages/stellar_client/lib/src/client.dart

```diff
@@ -473,4 +509,263 @@ class Client {
+      required String buyingAssetCode,
+      required String amount,
+      required String price,
+      String? memo}) async {
+    if (!_currencies.currencies.containsKey(sellingAssetCode)) {
+      throw Exception('Sell asset $sellingAssetCode is not available.');
+    }
+    if (!_currencies.currencies.containsKey(buyingAssetCode)) {
+      throw Exception('Buy asset $buyingAssetCode is not available.');
+    }
+
+    late final Asset sellAsset;
+    late final Asset buyAsset;
+
+    if (sellingAssetCode == 'XLM') {
+      sellAsset = AssetTypeNative();
+    } else {
+      sellAsset = AssetTypeCreditAlphaNum4(
+          _currencies.currencies[sellingAssetCode]!.assetCode,
+          _currencies.currencies[sellingAssetCode]!.issuer);
+    }
+    if (buyingAssetCode == 'XLM') {
+      buyAsset = AssetTypeNative();
+    } else {
+      buyAsset = AssetTypeCreditAlphaNum4(
+          _currencies.currencies[buyingAssetCode]!.assetCode,
+          _currencies.currencies[buyingAssetCode]!.issuer);
+    }
+
+    final ManageBuyOfferOperation buyOfferOperation =
+        ManageBuyOfferOperationBuilder(sellAsset, buyAsset, amount, price)
+            .build();
+
+    final account = await _sdk.accounts.account(accountId);
+    final balances = account.balances;
+
+    try {
+      final sellAssetBalance = balances.firstWhere(
+        (balance) {
+          if (sellingAssetCode == 'XLM' && balance.assetCode == null) {
+            // Special case for XLM
+            return true;
+          } else {
+            return balance.assetCode == sellingAssetCode;
+          }
+        },
+        orElse: () {
+          logger.e("Sell asset $sellingAssetCode not found in balances.");
+          throw Exception('Insufficient balance in $sellingAssetCode');
+        },
+      );
+
+      final double sellAmount = double.parse(amount);
+      final double availableBalance = double.parse(sellAssetBalance.balance);
+
+      if (sellAmount > availableBalance) {
+        throw Exception(
+            'Insufficient balance in $sellingAssetCode. Available: $availableBalance');
+      }
+    } catch (e) {
+      logger.e("Error: ${e.toString()}");
+      rethrow;
+    }
+
+    final Transaction transaction = TransactionBuilder(account)
+        .addOperation(buyOfferOperation)
+        .addMemo(memo != null ? Memo.text(memo) : Memo.none())
+        .build();
+
+    transaction.sign(_keyPair, _stellarNetwork);
+    try {
+      final SubmitTransactionResponse response =
+          await _sdk.submitTransaction(transaction);
+      if (!response.success) {
+        logger.e('Transaction failed with result: ${response.resultXdr}');
+      }
+      return response;
+    } catch (error) {
+      throw Exception('Transaction failed due to: ${error.toString()}');
+    }
+  }
+
+  Future<SubmitTransactionResponse> cancelOrder(
+      {required String sellingAssetCode,
+      required String buyingAssetCode,
+      required String offerId,
```

Comment:
How can I get the offer ID? It isn't included in the order response or the order book.